packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1321,8 +1321,19 @@ class PubspecDependency extends PubspecLine {
 
   /// This generates the entry for this dependency for the pubspec.yaml for the
   /// fake package that we'll use to get the version numbers figured out.
-  void describeForFakePubspec(StringBuffer dependencies, StringBuffer overrides, { bool useAnyVersion = true}) {
-    final String versionToUse = useAnyVersion || version.isEmpty ? 'any' : version;
+  void describeForFakePubspec(StringBuffer dependencies, StringBuffer overrides, { bool doUpgrade = true }) {
```

Comment:
Please add some comments describing the behavior with and without `doUpgrade`.

Reply:
Good idea, this file is like black magic...

Reply:
Done